renderer/src/components/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,245 @@
+import { parseReport, serializeReport, ReportAndNotes } from "@/helpers/utils/reports";
+import useScreenSizes from "@/helpers/hooks/useScreenSizes";
+import { IPC_MAIN_CHANNELS } from "@electron/helpers/constants";
+import { SCREENS } from "@/constants";
+import { MainPageProps } from "./types";
+
+const MainPage = ({
+  selectedDate,
+  selectedDateActivities,
+  setSelectedDateActivities,
+  shouldAutosave,
+  setShouldAutosave,
+  setSelectedDate,
+  latestProjAndAct,
+  setTrackTimeModalActivity,
+}: MainPageProps) => {
+  const [isDropboxConnected, setIsDropboxConnected] = useState(true);
+  const [calendarDate, setCalendarDate] = useState(new Date());
+  const [reportAndNotes, setReportAndNotes] = useState<any[] | ReportAndNotes>([]);
+  const [selectedDateReport, setSelectedDateReport] = useState("");
+  const { screenSizes } = useScreenSizes();
+  const isManualInputMain = localStorage.getItem("is-manual-input-main-section") === "true";
+  const showBookings = !!JSON.parse(localStorage.getItem("timetracker-user"));
+  const [reportsFolder] = useMainStore((state) => [state.reportsFolder, state.setReportsFolder], shallow);
+  const [isBeta] = useBetaStore((state) => [state.isBeta, state.setIsBeta], shallow);
+
+  useEffect(() => {
+    global.ipcRenderer.send(IPC_MAIN_CHANNELS.START_FOLDER_WATCHER, reportsFolder);
+    global.ipcRenderer.send(IPC_MAIN_CHANNELS.CHECK_DROPBOX_CONNECTION);
+    if (reportsFolder) {
+      global.ipcRenderer.on(IPC_MAIN_CHANNELS.CHECK_DROPBOX_CONNECTION, (event, data) => {
+        setIsDropboxConnected(!reportsFolder.includes("Dropbox") || data);
+      });
+    }
+    global.ipcRenderer.send(IPC_MAIN_CHANNELS.BETA_CHANNEL, isBeta);
+
+    return () => {
+      global.ipcRenderer.removeAllListeners(IPC_MAIN_CHANNELS.CHECK_DROPBOX_CONNECTION);
+      global.ipcRenderer.send(IPC_MAIN_CHANNELS.STOP_PATH_WATCHER, reportsFolder);
+    };
+  }, []);
+
+  useEffect(() => {
+    try {
+      if (shouldAutosave) {
+        const serializedReport =
+          serializeReport(selectedDateActivities) +
+          (!reportAndNotes[1] || reportAndNotes[1].startsWith("undefined") ? "" : reportAndNotes[1]);
+
+        saveSerializedReport(serializedReport);
+        setShouldAutosave(false);
+      }
+    } catch (err) {
+      global.ipcRenderer.send(
+        IPC_MAIN_CHANNELS.FRONTEND_ERROR,
+        "Reports saving error",
+        "An error occurred while saving reports. ",
+        err,
+      );
+    }
+  }, [selectedDateActivities]);
+
+  useEffect(() => {
+    if (selectedDateReport?.length > 0) {
+      const parsedReportsAndNotes = parseReport(selectedDateReport);
+      const parsedActivities = parsedReportsAndNotes[0];
+
+      setReportAndNotes(parsedReportsAndNotes);
+      setSelectedDateActivities(parsedActivities);
+      return;
+    }
+
+    setReportAndNotes([]);
+    setSelectedDateActivities([]);
+  }, [selectedDateReport]);
+
+  useEffect(() => {
+    (async () => {
```

Comment:
May I ask you to move the logic from the useEffect hook to a new function and invoke the function inside that useEffect hook? Name the function with a sense, please.